### PR TITLE
some minor changes to repoquery

### PR DESCRIPTION
**Pull request by max9631**
Do not merge! 
This PR is just infomative for jan silhan. Also I will be making some changes to it

### CODE REVIEW

**Comment by jsilhan** [comment]
> This PR is just infomative for jan silhan. Also I will be making some changes to it

@max9631, So what is missing to be ready for merge? This PR fixes the sorting issue, right?

I like any commit that removes more lines than adds and in the meantime provides new functionality.
@MichaelMraka knows repoquery code better than me. What's your opinion?

File: plugins/repoquery.py [hunk 1]
```diff
@@ -348,19 +378,46 @@ def run(self, args):
...
+            pkgs = set()
+            for pkg in providers.latest().run():
+                po = PackageWrapper(pkg)
+                try:
+                    pkgs.add(fmt_fn(po))
+                except AttributeError as e:
+                    # catch that the user has specified attributes
+                    # there don't exist on the dnf Package object.
+                    raise dnf.exceptions.Error(str(e))
+        for pkg in sorted(pkgs, key=lambda s: str(s)):
```
Comment (jsilhan) [inline plugins/repoquery.py]: @max9631 , change it to `for pkg in sorted(pkgs, key=i18n.ucd)` and see if it works.

**Comment by jsilhan** [comment]
> I am just wondering whether we should not keep the default behavior of repoquery here - to treet no package parameter supplied as a star glob e.g. "dnf repoquery --installed" == "dnf repoquery --installed * ". What do you think @jsilhan? 

Michal is right, the behavior of list switches should stay same as with any other filter option.

**Comment by mluscon** [comment]
I have just discovered that dnf repoquery  --qf "%{conflicts}" outputs: 

[<_hawkey.Reldep object, id: 62629>]
[<_hawkey.Reldep object, id: 64495>]
[<_hawkey.Reldep object, id: 68938>]
[<_hawkey.Reldep object, id: 69211>]
[<_hawkey.Reldep object, id: 69213>]
[<_hawkey.Reldep object, id: 69671>]
[<_hawkey.Reldep object, id: 69673>]
[<_hawkey.Reldep object, id: 69674>]
[<_hawkey.Reldep object, id: 70469>]
[<_hawkey.Reldep object, id: 73390>]
[<_hawkey.Reldep object, id: 73391>]

**Comment by mluscon** [comment]
Another issue spotted "dnf repoquery --upgrades": 

Traceback (most recent call last):
  File "/home/mluscon/projects/dnf/bin/dnf-2", line 36, in <module>
    main.user_main(sys.argv[1:], exit_code=True)
  File "/home/mluscon/projects/dnf/dnf/cli/main.py", line 193, in user_main
    errcode = main(args)
  File "/home/mluscon/projects/dnf/dnf/cli/main.py", line 84, in main
    return _main(base, args)
  File "/home/mluscon/projects/dnf/dnf/cli/main.py", line 139, in _main
    cli.run()
  File "/home/mluscon/projects/dnf/dnf/cli/cli.py", line 1083, in run
    return self.command.run(self.base.extcmds)
  File "/usr/lib/python2.7/site-packages/dnf-plugins/repoquery.py", line 296, in run
    for pkg in sorted(_list):
  File "/home/mluscon/projects/dnf/dnf/yum/misc.py", line 219, in __iter__
    return iter(self[self.__iter])
  File "/home/mluscon/projects/dnf/dnf/yum/misc.py", line 225, in __getitem__
    raise KeyError(item)
KeyError: u'upgrades'

File: plugins/repoquery.py [hunk 2]
```diff
@@ -276,14 +292,27 @@ def run(self, args):
             print(self.parser.format_help())
             return
 
+        if self.opts.list:
+            key = self.opts.key if len(self.opts.key) > 0 else None
+            _list = self.base.returnPkgLists(self.opts.list, key)
```
Comment (jsilhan) [inline plugins/repoquery.py]: Change it to `base.*_pkgs`, please.

**Comment by mluscon** [comment]
Retest this please

File: plugins/repoquery.py [hunk 3]
```diff
@@ -162,9 +164,24 @@ def parse_arguments(args):
...
+
+    help_list = {
+        'available': _('Display details about a available package or group of packages'),
+        'installed': _('Display details about a installed package or group of packages'),
+        'extras': _('Display details about a extra package or group of packages'),
+        'upgrades': _('Display details about a package upgrade or group of package upgrades'),
+        'autoremove': _('Display details about a autoremove package or group of packages'),
+        'recent': _('Display details about a recently edited package or group of packages'),
+        'downgrades': _('Display details about a downgrade package or group of packages')
+    }
```
Comment (jsilhan) [inline plugins/repoquery.py]: Reuse the help message from dnf list commands, please

**Comment by mluscon** [comment]
Merged, thanks a lot.

**Pull request closed without merging**